web/ui/mantine-ui/src/components/NotificationsProvider.tsx

```diff
@@ -1,28 +1,32 @@
-import React, { useEffect, useState, useCallback } from 'react';
+import React, { useEffect, useState } from 'react';
 import { useSettings } from '../state/settingsSlice';
 import { NotificationsContext } from '../state/useNotifications';
 import { Notification, NotificationsResponse } from "../api/responseTypes/notifications";
+import { useAPIQuery } from '../api/api';
 
 export const NotificationsProvider: React.FC<{ children: React.ReactNode }> = ({ children }) => {
   const { pathPrefix } = useSettings();
   const [notifications, setNotifications] = useState<Notification[]>([]);
   const [isConnectionError, setIsConnectionError] = useState(false);
   const [shouldFetchFromAPI, setShouldFetchFromAPI] = useState(false);
 
-  // Function to manually fetch notifications from the API, when Server-Side-Events don't work.
-  const fetchNotifications = useCallback(async () => {
-    try {
-      const response = await fetch(`${pathPrefix}/api/v1/notifications`);
-      if (!response.ok) {
-        throw new Error('Network response was not ok');
-      }
-      const data: NotificationsResponse = await response.json();
-      setNotifications(data.data);
-      setIsConnectionError(false);
-    } catch (error) {
-      setIsConnectionError(true);
+  const { data, isError } = useAPIQuery<NotificationsResponse>({
+    path: '/notifications',
+    enabled: shouldFetchFromAPI,
+    refetchInterval: 10000,
+  });
+
+  useEffect(() => {
+    if (data && data.data) {
+      // TypeScript throws an error when directly casting `data.data` to `Notification[]`
+      // because it assumes `data` is the entire `NotificationsResponse` object, not just the `data` field.
+      // However, debugging shows that `data.data` is correctly structured as a `Notification[]`, so it seems
+      // the compiler is being overly strict here. By casting it to `unknown` and then to `Notification[]`,
+      // we resolve the error and make it clear that `data.data` is the array of notifications, not the full response.
+      setNotifications(data.data as unknown as Notification[]);
```

Comment:
The issue here is that the `NotificationsResponse` type is inconsistently defined compared to the other ones in https://github.com/prometheus/prometheus/tree/main/web/ui/mantine-ui/src/api/responseTypes, see for example https://github.com/prometheus/prometheus/blob/main/web/ui/mantine-ui/src/api/responseTypes/tsdbStatus.ts.

The other types are structured in such a way that they already only represent the `data` field of the API result, and thus they also are called `...Result` and not `...Response`. If you make the type consistent with the others (and call it `NotificationsResult`), then things should be fine here.